java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -513,6 +514,72 @@ void testReadParquetFull() {
     }
   }
 
+  @Test
+  void testReadORC() {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    ORCOptions opts = ORCOptions.builder()
+            .includeColumn("float1")
+            .includeColumn("boolean1")
+            .includeColumn("byte1")
+            .build();
+    try (Table table = Table.readORC(opts, TEST_ORC_FILE)) {
+      long rows = table.getRowCount();
+      assertEquals(2, rows);
+      assertTableTypes(new DType[]{DType.FLOAT32, DType.BOOL8, DType.INT8}, table);
+    }
+  }
+
+  @Test
+  void testReadORCBuffer() throws IOException {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    ORCOptions opts = ORCOptions.builder()
+            .includeColumn("float1")
+            .includeColumn("boolean1")
+            .includeColumn("byte1")
+            .build();
+
+    byte[] buffer = new byte[(int) TEST_ORC_FILE.length() + 1024];
+    int bufferLen = 0;
+    try (FileInputStream in = new FileInputStream(TEST_ORC_FILE)) {
+      bufferLen = in.read(buffer);
+    }
+    try (Table table = Table.readORC(opts, buffer, 0, bufferLen)) {
+      long rows = table.getRowCount();
+      assertEquals(2, rows);
+      assertTableTypes(new DType[]{DType.FLOAT32, DType.BOOL8, DType.INT8}, table);
+    }
+  }
+
+  @Test
+  void testReadORCFull() {
+    assumeTrue(Cuda.isEnvCompatibleForTesting());
+    try (Table table = Table.readORC(TEST_ORC_FILE)) {
+      long rows = table.getRowCount();
+      assertEquals(2, rows);
+
+      DType[] expectedTypes = new DType[]{
+              DType.BOOL8,
+              DType.INT8,
+              DType.INT16,
+              DType.INT32,
+              DType.INT64,
+              DType.FLOAT32,
+              DType.FLOAT64,
+              DType.STRING,
+              DType.STRING,
+              DType.INT32,
+              DType.STRING,
+              DType.INT32,
+              DType.STRING,
+              DType.STRING,
+              DType.INT32,
+              DType.STRING
```

Comment:
This schema doesn't match what I'm seeing for the ORC data.  According to Spark, this file has a schema of:
```
StructField(boolean1,BooleanType,true)
StructField(byte1,ByteType,true)
StructField(short1,ShortType,true)
StructField(int1,IntegerType,true)
StructField(long1,LongType,true)
StructField(float1,FloatType,true)
StructField(double1,DoubleType,true)
StructField(bytes1,BinaryType,true)
StructField(string1,StringType,true)
StructField(middle,StructType(StructField(list,ArrayType(StructType(StructField(int1,IntegerType,true), StructField(string1,StringType,true)),true),true)),true)
StructField(list,ArrayType(StructType(StructField(int1,IntegerType,true), StructField(string1,StringType,true)),true),true)
StructField(map,MapType(StringType,StructType(StructField(int1,IntegerType,true), StructField(string1,StringType,true)),true),true)
```

cudf doesn't support compound vectors like lists or maps, so curious how this is working.  Is the compound nature being conveyed in the resulting column names or something similar?  Anyway I'm not sure this is the best test file given that I could see cudf supporting some of these compound results in the future which will change the resulting DType and thus fail this test when that occurs.

Reply:
I had taken orc file from cudf cpp tests. This is the first time I am using ORC file. Have removed the columns which has list,map etc and updated the unit tests.